makefile

```diff
@@ -629,6 +629,15 @@ install: $(if $(static),$(BLD)libmfem.a) $(if $(shared),$(BLD)libmfem.$(SO_EXT))
 	   ( $(MKINSTALLDIR) $(PREFIX_INC)/mfem/$$dir ) && \
 	   $(INSTALLDEF) $(SRC)$$dir/*.hpp $(PREFIX_INC)/mfem/$$dir; \
 	done
+# install miniapp common libraries and headers if built
+	if ls $(BLD)miniapps/common/libmfem-common.* &> /dev/null; then \
+	   $(if $(static),$(INSTALLDEF) $(BLD)miniapps/common/libmfem-common.a $(PREFIX_LIB) &&) \
+	   $(if $(shared),$(INSTALLDEF) $(BLD)miniapps/common/libmfem-common.$(SO_VER) $(PREFIX_LIB) \
+	      && chmod $(INSTALL_BIN_PERM) $(PREFIX_LIB)/libmfem-common.$(SO_VER) \
+	      && ln -sf libmfem-common.$(SO_VER) $(PREFIX_LIB)/libmfem-common.$(SO_EXT) &&) \
+	   $(MKINSTALLDIR) $(PREFIX_INC)/mfem/miniapps/common && \
+	   $(INSTALLDEF) $(SRC)miniapps/common/*.hpp $(PREFIX_INC)/mfem/miniapps/common; \
+	fi
```

Comment:
In the GNU make build system, we're not installing examples, so I prefer to not install miniapps either. Unless, of course, you actually want to use this feature.

Reply:
This wouldn't be installing binaries, only libraries and headers for the common miniapp, so I'd argue it's a bit different. If you are open to the idea of merging this, I'll open another PR though, I don't want to overload the discussion here, I already feel guilty enough for suggesting `GNUInstallDirs`.

Reply:
If you think you are going to use this, it's not a problem to include it. I just thought you are primarily interested in the CMake build system.

One suggestion: I think it is better to trigger this installation explicitly via a makefile variable, e.g. `MFEM_INSTALL_MINIAPPS_COMMON`, set to `YES/NO` with a default setting in `config/defaults.mk`. The variable can be set during config time and exported to `config.mk` similar to other configuration options. I can make this change, if you are not familiar with all the details on how to do this.

Reply:
I'm happy to live with CMake personally and for our purposes it does everything we need.

I was just trying to go for feature parity (just like #4252 was).
I guess I perhaps don't understand why we are maintaining two build systems, I thought it was for the convenience of the user, to allow them to use whatever they are most comfortable with. For instance, the "Getting Started/Building MFEM" manual page uses GNU make, and doesn't even mention CMake, but then the user is locked out of some functionality if they don't migrate to CMake.

Reply:
> I perhaps don't understand why we are maintaining two build systems

Different users use different build systems, so we don't want to force them to switch.

> "Getting Started/Building MFEM" manual page uses GNU make, and doesn't even mention CMake

This will be addressed in https://github.com/mfem/web/pull/291. 😄 

> the user is locked out of some functionality if they don't migrate to CMake

There are only minor differences between GNU make and CMake functionality -- one example of that being that installing the MFEM examples is not supported by GNU make.

Regarding the current PR: I'm fine either way: add or don't add the GNU make support for installing the `mfem-common` library and its headers.

Reply:
> I think it is better to trigger this installation explicitly via a makefile variable, e.g. MFEM_INSTALL_MINIAPPS_COMMON, set to YES/NO with a default setting in config/defaults.mk.

I agree doing a raw `ls` to check for existence is a bit rough. And I think defining such a variable goes a bit against the philosophy of our GNU makefiles as you've hinted at. I think the cleanest, most natural solution would be to define a new rule such as `install-miniapps: install miniapps` which would only work in practice for people who actually open and inspect the makefile (in the same way as, e.g., the `miniapps` or `test-miniapps` rules which we don't publicise to the user). So, if the team prefers to just not have the option, I'm happy to drop it.